### PR TITLE
Add resources to setup ephemeral Ruby repositories

### PR DESCRIPTION
**Pull request by tichavskym**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a OpenAPI schema is updated (if applicable)
- n/a DB schema change has corresponding DB migration (if applicable)
- n/a README updated (if worker configuration changed, or if applicable)

### CODE REVIEW

File: tests/test_workers/test_pkg_managers/test_ruby.py
Comment (tichavskym) [reply tests/test_workers/test_pkg_managers/test_ruby.py]: Yes, you're right

**Comment by tichavskym** [comment]
I accidentally pushed commit to this branch, then dropped it, but the pipeline stopped working. It looks like [Github Runner issue](https://github.com/actions/runner/issues/1923) though.

**Review by lkolacek — APPROVED** [review]

**Comment by brunoapimentel** [comment]
Sorry for the late comment, and it's probably a small detail, but I was wondering about the names we're choosing here.

The files and Nexus repos are using the `ruby` name, which is the language name, whereas all other use the package manager name, that is: pip instead of python, npm or yarn instead of nodejs, gomod instead of golang. (only oddball is git-submodule)

So, should we change the names to `gem` instead of `ruby`. I think it would follow the standard we have here more closely, although, IIUIC, Gem is the only package manager for Ruby, so maybe changing names would not bring much benefit.

Wdyt @chmeliik @lkolacek @tichavskym ?

**Comment by taylormadore** [comment]
Perhaps consider using rubygem for the names?

**Comment by chmeliik** [comment]
Hmm, good point. Which package manager are we actually adding here? [rubygems](https://rubygems.org/pages/download)? [bundler](https://bundler.io/)? both?

**Comment by tichavskym** [comment]
We'll be using `Gemfile.lock` files generated by `bundler` to identify which gems (which we're at some point created and uploaded to the server by `RubyGems` package manager = `gem` CLI tool) to download from `rubygems.org`.

**Comment by brunoapimentel** [comment]
Looking from a noobie perspective, I would guess that bundler is a tool on top of rubygems in a similar way that pip-compile works on top of pip. So I'm thinking we could go with `rubygems`, as @taylormadore suggested.

**Review by chmeliik — APPROVED** [review]

**Comment by tichavskym** [comment]
I consider this pull request to be ready to merge, so if there's anything you expect from me (rebasing it, merging it), please let me know.

**Comment by chmeliik** [comment]
> I consider this pull request to be ready to merge, so if there's anything you expect from me (rebasing it, merging it), please let me know.

We typically squash "fixup" type commits to the original commit, the 3 commits feel like they could be a single one. After that, one of us maintainers can merge this :+1:

**Pull request merged**
